### ISSUE TITLE
Trying to build but submodules fail

### ISSUE THREAD

**Issue by TipVFL** [syl22-00/pocketsphinx.js#135]
I tried to follow the directions from the readme, but when I got to the "git submodule update" it doesn't work and gives me "Permission denied (publickey). fatal: Could not read from remote repository."

If I try cloning pocketsphinx.js with --recursive I get the same error. I tried just cloning and setting up the submodules myself, but I could not get it working, I always ended up with multiple undefined symbols during the final build:
> warning: undefined symbol: __syscall140
> warning: undefined symbol: __syscall145
> warning: undefined symbol: popen

It still generates the JS and WASM files, but they don't work correctly and immediately error out in the browser saying that the function "__syscall145" doesn't exist, which was one of the undefined symbols from the build. 

I've been trying all day to get this to work, but I'm stuck. 

Here's the full text I get when I try to "submodule update"
> Cloning into '/downloads/pocketsphinx.js/pocketsphinx'...
> git@github.com: Permission denied (publickey).
> fatal: Could not read from remote repository.
> 
> Please make sure you have the correct access rights
> and the repository exists.
> fatal: clone of 'git@github.com:cmusphinx/pocketsphinx.git' into submodule path '/downloads/pocketsphinx.js/pocketsphinx' failed
> Failed to clone 'pocketsphinx'. Retry scheduled
> Cloning into '/downloads/pocketsphinx.js/sphinxbase'...
> git@github.com: Permission denied (publickey).
> fatal: Could not read from remote repository.
> 
> Please make sure you have the correct access rights
> and the repository exists.
> fatal: clone of 'git@github.com:cmusphinx/sphinxbase.git' into submodule path '/downloads/pocketsphinx.js/sphinxbase' failed
> Failed to clone 'sphinxbase'. Retry scheduled
> Cloning into '/downloads/pocketsphinx.js/pocketsphinx'...
> git@github.com: Permission denied (publickey).
> fatal: Could not read from remote repository.
> 
> Please make sure you have the correct access rights
> and the repository exists.
> fatal: clone of 'git@github.com:cmusphinx/pocketsphinx.git' into submodule path '/downloads/pocketsphinx.js/pocketsphinx' failed
> Failed to clone 'pocketsphinx' a second time, aborting

It does build and produce a pocketsphinx.js and a pocketsphinx.wasm but when I replace those files in the examples I just get these errors:

> pocketsphinx.js:1 missing function: __syscall145
> ___syscall145 @ pocketsphinx.js:1
> (anonymous) @ wasm-001f1032-2069:1
> (anonymous) @ wasm-001f1032-424:1
> (anonymous) @ wasm-001f1032-254:1
> (anonymous) @ wasm-001f1032-942:1
> (anonymous) @ wasm-001f1032-1219:1
> (anonymous) @ wasm-001f1032-1220:1
> (anonymous) @ wasm-001f1032-1213:1
> (anonymous) @ wasm-001f1032-444:1
> (anonymous) @ wasm-001f1032-2032:1
> (anonymous) @ wasm-001f1032-1395:1
> (anonymous) @ wasm-001f1032-897:1
> (anonymous) @ wasm-001f1032-1467:1
> Module.dynCall_iii @ pocketsphinx.js:1
> dynCall_iii_62 @ VM124:4
> constructor_body @ pocketsphinx.js:1
> (anonymous) @ pocketsphinx.js:1
> Recognizer @ VM122:4
> initialize @ recognizer.js:136
> (anonymous) @ recognizer.js:63
> pocketsphinx.js:1 -1
> pocketsphinx.js:1 -1
> abort @ pocketsphinx.js:1
> ___syscall145 @ pocketsphinx.js:1
> (anonymous) @ wasm-001f1032-2069:1
> (anonymous) @ wasm-001f1032-424:1
> (anonymous) @ wasm-001f1032-254:1
> (anonymous) @ wasm-001f1032-942:1
> (anonymous) @ wasm-001f1032-1219:1
> (anonymous) @ wasm-001f1032-1220:1
> (anonymous) @ wasm-001f1032-1213:1
> (anonymous) @ wasm-001f1032-444:1
> (anonymous) @ wasm-001f1032-2032:1
> (anonymous) @ wasm-001f1032-1395:1
> (anonymous) @ wasm-001f1032-897:1
> (anonymous) @ wasm-001f1032-1467:1
> Module.dynCall_iii @ pocketsphinx.js:1
> dynCall_iii_62 @ VM124:4
> constructor_body @ pocketsphinx.js:1
> (anonymous) @ pocketsphinx.js:1
> Recognizer @ VM122:4
> initialize @ recognizer.js:136
> (anonymous) @ recognizer.js:63
> pocketsphinx.js:1 Uncaught abort(-1). Build with -s ASSERTIONS=1 for more info.
> abort @ pocketsphinx.js:1
> ___syscall145 @ pocketsphinx.js:1
> (anonymous) @ wasm-001f1032-2069:1
> (anonymous) @ wasm-001f1032-424:1
> (anonymous) @ wasm-001f1032-254:1
> (anonymous) @ wasm-001f1032-942:1
> (anonymous) @ wasm-001f1032-1219:1
> (anonymous) @ wasm-001f1032-1220:1
> (anonymous) @ wasm-001f1032-1213:1
> (anonymous) @ wasm-001f1032-444:1
> (anonymous) @ wasm-001f1032-2032:1
> (anonymous) @ wasm-001f1032-1395:1
> (anonymous) @ wasm-001f1032-897:1
> (anonymous) @ wasm-001f1032-1467:1
> Module.dynCall_iii @ pocketsphinx.js:1
> dynCall_iii_62 @ VM124:4
> constructor_body @ pocketsphinx.js:1
> (anonymous) @ pocketsphinx.js:1
> Recognizer @ VM122:4
> initialize @ recognizer.js:136
> (anonymous) @ recognizer.js:63

I've also tried building it all in the JS file, and I get a very similar set of errors:

> missing function: __syscall145
> ___syscall145 @ pocketsphinx.js:1
> NG @ pocketsphinx.js:10
> sH @ pocketsphinx.js:10
> hI @ pocketsphinx.js:10
> xE @ pocketsphinx.js:6
> kv @ pocketsphinx.js:7
> jv @ pocketsphinx.js:7
> yv @ pocketsphinx.js:7
> $f @ pocketsphinx.js:15
> Lg @ pocketsphinx.js:15
> yd @ pocketsphinx.js:15
> Ig @ pocketsphinx.js:15
> u7 @ pocketsphinx.js:16
> dynCall_iii_62 @ VM122:4
> constructor_body @ pocketsphinx.js:1
> (anonymous) @ pocketsphinx.js:1
> Recognizer @ VM120:4
> initialize @ recognizer.js:86
> (anonymous) @ recognizer.js:23
> pocketsphinx.js:1 -1
> pocketsphinx.js:1 -1
> abort @ pocketsphinx.js:1
> ___syscall145 @ pocketsphinx.js:1
> NG @ pocketsphinx.js:10
> sH @ pocketsphinx.js:10
> hI @ pocketsphinx.js:10
> xE @ pocketsphinx.js:6
> kv @ pocketsphinx.js:7
> jv @ pocketsphinx.js:7
> yv @ pocketsphinx.js:7
> $f @ pocketsphinx.js:15
> Lg @ pocketsphinx.js:15
> yd @ pocketsphinx.js:15
> Ig @ pocketsphinx.js:15
> u7 @ pocketsphinx.js:16
> dynCall_iii_62 @ VM122:4
> constructor_body @ pocketsphinx.js:1
> (anonymous) @ pocketsphinx.js:1
> Recognizer @ VM120:4
> initialize @ recognizer.js:86
> (anonymous) @ recognizer.js:23
> pocketsphinx.js:1 Uncaught abort(-1). Build with -s ASSERTIONS=1 for more info.
> abort @ pocketsphinx.js:1
> ___syscall145 @ pocketsphinx.js:1
> NG @ pocketsphinx.js:10
> sH @ pocketsphinx.js:10
> hI @ pocketsphinx.js:10
> xE @ pocketsphinx.js:6
> kv @ pocketsphinx.js:7
> jv @ pocketsphinx.js:7
> yv @ pocketsphinx.js:7
> $f @ pocketsphinx.js:15
> Lg @ pocketsphinx.js:15
> yd @ pocketsphinx.js:15
> Ig @ pocketsphinx.js:15
> u7 @ pocketsphinx.js:16
> dynCall_iii_62 @ VM122:4
> constructor_body @ pocketsphinx.js:1
> (anonymous) @ pocketsphinx.js:1
> Recognizer @ VM120:4
> initialize @ recognizer.js:86
> (anonymous) @ recognizer.js:23

Any help getting this to build would be hugely appreciated.

**Comment by bene51** [comment]
I had the same issue. I edited .gitmodules and replaced 
url = git@github.com:cmusphinx/sphinxbase.git
url = git@github.com:cmusphinx/pocketsphinx.git
with
url = https://github.com/cmusphinx/sphinxbase.git
url = https://github.com/cmusphinx/pocketsphinx.git

and afterwards run
$ git submodule init
$ git submodule update

Compilation then kind of worked (maybe not), but I don't get the erros you get, but a BindingError when I open the sample webapp in the browser

Let me know if you get any further...

**Comment by syl22-00** [comment]
Right, the assumption is that you connect to github via ssh (https://help.github.com/en/github/authenticating-to-github/connecting-to-github-with-ssh). If you want to connect via HTTPS, you will need to update `.gitmodules`.
